### PR TITLE
HOCS-5281: migrate field definitions to service

### PR DESCRIPTION
**Pull request by LiamMacP**
Refactor of the screens used within the system to bring them into the service. This effectively removes the complex SQL changes currently required to propagate a screen change. This simplifies the system by allowing easy changes to be handled within JSON files and patched as necessary.

To aid this a JsonFileReader has been added, this at runtime allows for screens to be loaded by the system and using the current infrastructure return the same response payload as before (supporting backwards compatibility).

### CODE REVIEW

File: src/main/resources/screens/COMP_HIDDEN.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
```
Comment (LiamMacP) [inline src/main/resources/screens/COMP_HIDDEN.json:1]: Screens like these aren't actually required anymore as summary, accordion and extracts all have their own config. Removing this aren't in scope of this change though - as we need to ensure each isn't used _somewhere_...

File: src/main/java/uk/gov/digital/ho/hocs/workflow/domain/repositories/JsonFileReader.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.workflow.application.LogEvent;
+import uk.gov.digital.ho.hocs.workflow.domain.exception.ApplicationExceptions;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+@Service
+public abstract class JsonFileReader {
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/workflow/domain/repositories/JsonFileReader.java:13]: I'm not actually convinced this is a useful helper - we could and probably should role this into the ScreenRepository

File: src/main/java/uk/gov/digital/ho/hocs/workflow/domain/repositories/JsonFileReader.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        this.objectMapper = objectMapper;
+    }
+
+    protected <T> T readValueFromFile(TypeReference<T> reference, String fileName) {
+        try (InputStream in = Thread.currentThread().getContextClassLoader().getResourceAsStream(
+            String.format("%s/%s.json", getFolderName(), fileName))) {
+            return objectMapper.readValue(in, reference);
+        } catch (IOException e) {
+            e.printStackTrace();
+            throw new ApplicationExceptions.JsonFileReadException(
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/workflow/domain/repositories/JsonFileReader.java:27]: This also supports moving into the ScreenRepository - we should have a proper Exception...

**Review by LiamMacP — COMMENTED** [review]
This is a very _dumb_ implementation - we could be cleverer with directories for the screens - merging with respective camunda diagrams. We would have to take care with shared screens - but symbolic links should help there. Out of scope for this though.

File: src/main/java/uk/gov/digital/ho/hocs/workflow/domain/model/forms/HocsFormField.java [hunk 4]
```diff
@@ -33,13 +35,26 @@ public static HocsFormField from(FieldDto fieldDto) {
         return new HocsFormField(fieldDto.getComponent(), fieldDto.getValidation(), props);
     }
 
+    public static HocsFormField from(Field field) {
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/workflow/domain/model/forms/HocsFormField.java:38]: This is a dupe of the above with just a different method parameter - we could be smarter but as it is deprecated I don't see the point.

File: src/main/java/uk/gov/digital/ho/hocs/workflow/domain/repositories/ScreenRepository.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import uk.gov.digital.ho.hocs.workflow.domain.repositories.entity.Schema;
+
+@Service
+public class ScreenRepository extends JsonFileReader {
+
+    protected ScreenRepository(ObjectMapper objectMapper) {
+        super(objectMapper);
+    }
+
+    //TODO: investigate cachable benefits
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/workflow/domain/repositories/ScreenRepository.java:15]: Having run this locally some of the files took around 0.02ms to load, with repeat loads of < 0.005ms. I'm not convinced caching is necessary for this at the moment.

**Review by SWhitlam — APPROVED** [review]

**Pull request merged**